lib/Frontend/Frontend.cpp

```diff
@@ -1168,9 +1168,10 @@ CompilerInstance::getSourceFileParsingOptions(bool forPrimary) const {
     opts |= SourceFile::ParsingFlags::SuppressWarnings;
   }
 
-  // Enable interface hash computation for primaries, but not in WMO, as it's
-  // only currently needed for incremental mode.
-  if (forPrimary) {
+  // Enable interface hash computation for primaries or emit-module-separately,
+  // but not in WMO, as it's only currently needed for incremental mode.
+  if (forPrimary ||
+      typeOpts.SkipFunctionBodies == FunctionBodySkipping::NonInlinableWithoutTypes) {
```

Comment:
Can we use `FrontOptions::ActionType::EmitModuleOnly` here instead?

Reply:
We can if we don't mind turning on interface hashing in some non-incremental builds. `EmitModuleOnly` is used when we skip compilation, it's already pretty common before the emit-module-separately work afaict. It looks like the stdlib is built this way via CMake and a lot of tests just do an `-emit-module` without an `-emit-library`, so they are considered `EmitModuleOnly`.

What do you think? Can we turn on interface hashing in more cases or should we keep limiting it only to emit-module-separately builds with the check on `-experimental-skip-non-inlinable-function-bodies-without-types` or some new flag?